src/posix/platform/system.hpp

```diff
@@ -38,3 +43,14 @@
  * @returns  If the system runs in dry-run mode.
  */
 bool IsSystemDryRun(void);
+
+/**
+ * @brief Triggers a hardware reset of the radio chip.
+ *
+ * overwrite this to use platform-specific function to trigger hardware reset
+ *
+ * @returns otError OT_ERROR_NOT_IMPLEMENTED means using default implementation
+ */
+extern "C" otError otSysHardwareReset(void);
```

Comment:
`otSys` API should be declared in https://github.com/openthread/openthread/blob/main/src/posix/platform/include/openthread/openthread-system.h.